http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/BackgroundProcess.scala

```diff
@@ -0,0 +1,92 @@
+package org.broadinstitute.dsde.workbench.leonardo.monitor
+
+import cats.Show
+import cats.effect.Async
+import cats.syntax.all._
+import fs2.Stream
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+import org.typelevel.log4cats.StructuredLogger
+
+import java.time.Instant
+import java.util.UUID
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * Defines a common interface that certain background processes should implement.
+ * This type of background processes do the following periodically:
+ * 1. Query the database for a list of candidates
+ * 2. Filter the candidates based on some criteria
+ * 3. Perform some action on the filtered candidates
+ */
+trait BackgroundProcess[F[_], A] {
+  def monitorType: String
+  def interval: FiniteDuration
+
+  /**
+   * Given a candidate, determine whether it should be filtered out.
+   * true: keep the candidate
+   * false: filter out the candidate
+   */
+  def filterCriteria(a: A, now: Instant)(implicit
+    F: Async[F],
+    metrics: OpenTelemetryMetrics[F],
+    logger: StructuredLogger[F]
+  ): F[Boolean]
+
+  /**
+   * Get a collection of candidates from the database
+   */
+  def dbSource(): F[Seq[A]]
```

Comment:
Ultimately this could be a list of candidates from a variety of sources ... does it make sense to name this as such?

Reply:
sounds good 👍